src/overlays/actors/ovl_En_Goma/z_en_goma.c

```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
+            Matrix_RotateZ(this->slopeRoll / 32768.0f * 3.1415927f, 1);
+            Matrix_RotateY(this->actor.shape.rot.y / 32768.0f * 3.1415927f, 1);
+            Matrix_RotateX(this->actor.shape.rot.x / 32768.0f * 3.1415927f, 1);
+            Matrix_RotateZ(this->actor.shape.rot.z / 32768.0f * 3.1415927f, 1);
+            Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, 1);
+            SkelAnime_DrawOpa(globalCtx, this->skelanime.skeleton, this->skelanime.jointTable, &EnGoma_OverrideLimbDraw,
+                              0, this);
+            break;
+
+        case ENGOMA_EGG:
+            this->actor.naviEnemyId = 2;
+            y = (s16)(sinf((this->eggTimer * 5.0f * 3.1415f) / 180.0f) * 31.9f);
+            y = (s16)(y + 31);
+            gSPSegment(POLY_OPA_DISP++, 0x08, func_80094E78(globalCtx->state.gfxCtx, 0, y));
+            Matrix_Push();
+            Matrix_Scale(this->eggScale, 1.0f / this->eggScale, this->eggScale, 1);
+            Matrix_RotateY(this->eggSquishAngle * 0.15f, 1);
+            Matrix_RotateZ(this->eggSquishAngle * 0.1f, 1);
+            Matrix_Scale(0.95f - this->eggSquishAmount, this->eggSquishAmount + 1.05f, 0.95f - this->eggSquishAmount,
+                         1);
+            Matrix_RotateZ(-(this->eggSquishAngle * 0.1f), 1);
+            Matrix_RotateY(-(this->eggSquishAngle * 0.15f), 1);
+            Matrix_Translate(0.0f, this->eggYOffset, 0.0f, 1);
+            Matrix_RotateX(this->eggPitch, 1);
+            gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_goma.c", 2101),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_OPA_DISP++, D_06002A70);
+            Matrix_Pop();
+            break;
+
+        case ENGOMA_HATCH_DEBRIS:
+            gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_goma.c", 2107),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_OPA_DISP++, gBrownFragmentDL);
+            break;
+
+        case ENGOMA_BOSSLIMB:
+            if (this->bossLimbDl != NULL) {
+                gSPSegment(POLY_OPA_DISP++, 0x08, EnGoma_NoBackfaceCullingDlist(globalCtx->state.gfxCtx));
+                gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_goma.c", 2114),
+                          G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                gSPDisplayList(POLY_OPA_DISP++, this->bossLimbDl);
+            }
+            break;
+    }
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_goma.c", 2119);
+}
+
+void EnGoma_Debris(EnGoma* this, GlobalContext* globalCtx) {
+    this->actor.shape.rot.y += 2500;
+    this->actor.shape.rot.x += 3500;
+    if (this->actionTimer == 0) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnGoma_SpawnHatchDebris(EnGoma* this, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
```

Comment:
```suggestion
void EnGoma_SpawnHatchDebris(EnGoma* this, GlobalContext* globalCtx2) {
    GlobalContext* globalCtx = globalCtx2;
```
And switch them in the functions that use them.